playback/tapeRecorder.ts

```diff
@@ -91,26 +123,60 @@ export class TapeRecorder {
       });
 
       if (this.inPlaybackMode()) {
-        throw new OperationExceptionDuringPlayback();
+        // In playback mode we want to capture this as an error that is an output of the function
+        //  which is a legit recorded result, and not fail the playback it self
+        throw generatePlaybackException(
+          PlaybackExceptionTypes.OperationExceptionDuringPlayback
+        );
       }
 
-      throw error;
+      return error;
+    };
+    const handleResult = (result: any): any => {
+      this.recordOutput({
+        alias: OPERATION_OUTPUT_ALIAS,
+        invocationNumber: 1,
+        args: [result],
+      });
+    };
+
+    let result: any;
+    try {
+      result = func(...args);
+    } catch (error) {
+      throw handleException(error);
     }
 
-    this.recordOutput({
-      alias: OPERATION_OUTPUT_ALIAS,
-      invocationNumber: 1,
-      args: [result],
-    });
+    if (result && typeof result.then === "function") {
+      result.then(
+        (realResult: any) => {
+          handleResult(realResult);
+
+          return realResult;
+        },
+        (err: any) => {
+          handleException(err);
```

Comment:
I think you are swallowing  the exception here 
original user will be unable to catch rejections 
`wrappedFn().then(...).catch((err)=> ...)`
catch will get the